proxy/http2/HPACK.cc

```diff
@@ -946,22 +922,25 @@ hpack_encode_header_block(HpackIndexingTable &indexing_table, uint8_t *out_buf,
 
   MIMEFieldIter field_iter;
   for (MIMEField *field = hdr->iter_get_first(&field_iter); field != nullptr; field = hdr->iter_get_next(&field_iter)) {
-    HpackField field_type;
-    MIMEFieldWrapper header(field, hdr->m_heap, hdr->m_http->m_fields_impl);
-    int name_len;
-    int value_len;
-    const char *name = header.name_get(&name_len);
-    header.value_get(&value_len);
+    std::string_view name  = field->name_get();
+    std::string_view value = field->value_get();
+
     // Choose field representation (See RFC7541 7.1.3)
     // - Authorization header obviously should not be indexed
     // - Short Cookie header should not be indexed because of low entropy
-    if ((ptr_len_casecmp(name, name_len, MIME_FIELD_COOKIE, MIME_LEN_COOKIE) == 0 && value_len < 20) ||
-        (ptr_len_casecmp(name, name_len, MIME_FIELD_AUTHORIZATION, MIME_LEN_AUTHORIZATION) == 0)) {
+    HpackField field_type;
+    // TODO: replace `strcasecmp` with `memcmp`
+    if ((strcasecmp(name, HPACK_HDR_FIELD_COOKIE) == 0 && value.size() < 20) ||
```

Comment:
If you're tuning, do the size comparison first.

Reply:
Should be. Thanks!

Reply:
Done.